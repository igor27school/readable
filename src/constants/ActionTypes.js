export const CHANGE_SORT_ORDER = "CHANGE_SORT_ORDER"
export const RECEIVE_CATEGORIES = "RECEIVE_CATEGORIES"
export const RECEIVE_ALL_POSTS = "RECEIVE_ALL_POSTS"
export const RECEIVE_CATEGORY_POSTS = "RECEIVE_CATEGORY_POSTS"
export const RECEIVE_POST = "RECEIVE_POST"
export const RECEIVE_POST_COMMENTS = "RECEIVE_POST_COMMENTS"
export const RECEIVE_VOTE = "RECEIVE_VOTE"
export const ADD_POST = "ADD_POST"
export const ADD_COMMENT = "ADD_COMMENT"
export const REMOVE_OBJECT = "REMOVE_OBJECT"
export const RECEIVE_COMMENT = "RECEIVE_COMMENT"
export const MODIFY_COMMENT = "MODIFY_COMMENT"
export const MODIFY_POST = "MODIFY_POST"
