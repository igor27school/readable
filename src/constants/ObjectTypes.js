// Identifies which object the operation is being performed on.
// Used for adding, deleting, editing, voting.
export const POST_TYPE = 'POST_TYPE'
export const COMMENT_TYPE = 'COMMENT_TYPE'
