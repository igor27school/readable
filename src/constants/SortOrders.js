export const SORT_BY_SCORES = "SORT_BY_SCORES"
export const SORT_BY_TIMESTAMPS = "SORT_BY_TIMESTAMPS"
