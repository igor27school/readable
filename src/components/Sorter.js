import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { reSort } from '../actions/ActionCreators'
import { SORT_BY_SCORES, SORT_BY_TIMESTAMPS } from '../constants/SortOrders'

/**
* @description This component is used in AllCategories, Category and PostDetails views. It allows the user to re-sort the posts and comments in a different order.
*/
export class Sorter extends Component {
  static propTypes = {
    sortOrder: PropTypes.oneOf([SORT_BY_SCORES, SORT_BY_TIMESTAMPS]).isRequired,
    reSort: PropTypes.func.isRequired,
  }
  render() {
    return (
      <select
        defaultValue={this.props.sortOrder}
        onChange={event => this.props.reSort(event.target.value)}
      >
        <option value={SORT_BY_SCORES}>Sort by highest score</option>
        <option value={SORT_BY_TIMESTAMPS}>Sort by most recent</option>
      </select>
    )
  }
}

function mapStateToProps ({ sortOrder }) {
  return {sortOrder}
}

function mapDispatchToProps (dispatch) {
  return {
    reSort: (sortBy) => dispatch(reSort(sortBy)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Sorter)
