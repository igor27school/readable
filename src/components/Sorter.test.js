import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { SORT_BY_TIMESTAMPS } from '../constants/SortOrders'
import { Sorter } from './Sorter'

jest.mock('../utils/ServerAPI')

describe('Sorter tests', () => {
  const setup = () => {
    const props = {
      sortOrder: SORT_BY_TIMESTAMPS,
      reSort: jest.fn(),
    }
    const renderer = new ShallowRenderer()
    return {
      props,
      renderer
    }
  }

  it('Sorter matches the snapshot', () => {
    const { renderer, props } = setup()
    renderer.render(<Sorter {...props}/>)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
