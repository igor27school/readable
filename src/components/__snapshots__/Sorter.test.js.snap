// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorter tests Sorter matches the snapshot 1`] = `
<select
  defaultValue="SORT_BY_TIMESTAMPS"
  onChange={[Function]}
>
  <option
    value="SORT_BY_SCORES"
  >
    Sort by highest score
  </option>
  <option
    value="SORT_BY_TIMESTAMPS"
  >
    Sort by most recent
  </option>
</select>
`;
